skynet/modules/ttt/summaries/processor.py

```diff
@@ -1,42 +1,50 @@
 from langchain.chains.summarize import load_summarize_chain
-from langchain.llms import LlamaCpp
-from langchain.prompts import PromptTemplate
+from langchain.prompts import ChatPromptTemplate
 from langchain.text_splitter import RecursiveCharacterTextSplitter
+from langchain_core.documents import Document
+from langchain_openai import ChatOpenAI
 
-from skynet.env import llama_n_batch, llama_n_gpu_layers, llama_path
+from skynet.env import openai_api_base_url, openai_api_key
+from skynet.logs import get_logger
 
-from .prompts.action_items import action_items_template
-from .prompts.summary import summary_template
+from .prompts.action_items import action_items_system_message
+from .prompts.summary import summary_system_message
 from .v1.models import Job, JobType
 
 llm = None
+map_reduce_threshold = 12000
+log = get_logger(__name__)
 
 
 def initialize():
     global llm
 
-    llm = LlamaCpp(
-        model_path=llama_path,
-        temperature=0.01,
-        max_tokens=4096,
-        n_ctx=4096,
-        n_gpu_layers=llama_n_gpu_layers,
-        n_batch=llama_n_batch,
+    llm = ChatOpenAI(
+        api_key=openai_api_key
+        or 'placeholder',  # if no key is set, use a placeholder value to bypass validation, and allow the custom base url to be used
+        base_url=None if openai_api_key else openai_api_base_url,
```

Comment:
Don't we need to pass headers here somehow? The OpenAI API app will also need the JWT.

Reply:
the openai app already has them

Reply:
I mean, if we get a summary API request, with a token, and we then need to contact the openai compat app how are we passing the token?

Reply:
Right!

Reply:
Headers can be set on the langchain llm object only when initialising it, doing it at a later time has no effect.

Thus, there are two ways of making this work that I can think of, given the current setup of our auth flow:

1. Initialise the llm object every time a request is processed and have the JWT be forwarded from the client facing request all the way through. This shouldn't be expensive since we're not loading the model anymore here.

2. Have a separate auth flow between summaries api and the our openai api services, and thus add a second type of auth validation on the openai api. 

There's also a third choice, where we get rid of JWTs and introduce service to service auth, if we'll go down that path we still need to figure out how and if we should forward the auth details between the summaries and our openai api services

Reply:
Perhaps we can fake it by using a class that implements "__str__" and can return different values per request.

It would only work if the headers are serialized late, but worth a shot IMHO.

Reply:
_ _ str _ _ (no spaces but GH will format it otherwise)

Reply:
there's no need for that, if we just wanna pass through the jwt, we'll just init a new llm object on each processing flow, it has no performance impact

Reply:
Ack. Let's go in this direction, since we want the app to be hit directly too, so we have to support tokens anyway.

Reply:
we need to discuss this a bit more, there are some other aspects to consider